### PR TITLE
Hide data constructors for FreeApplicative, expose only smart constru…

### PR DESCRIPTION
**Pull request by adelbertc**
…ctors

* Also rename run to foldMap to be consistent with Free

### CODE REVIEW

**Comment by adelbertc** [comment]
@mandubian

**Comment by mandubian** [comment]
Seems ok to me!

File: free/src/main/scala/cats/free/FreeApplicative.scala [hunk 1]
```diff
@@ -1,49 +1,56 @@
 package cats
 package free
 
+import cats.arrow.NaturalTransformation
+
 /** Applicative Functor for Free */
 sealed abstract class FreeApplicative[F[_], A] { self =>
-
-  import FreeApplicative.{FA, Pure, Ap}
+  import FreeApplicative.{FA, Pure, Ap, ap => apply, lift}
```
Comment (non) [inline free/src/main/scala/cats/free/FreeApplicative.scala]: I think this example would be a little bit more clear without aliasing `ap` to `apply`.

What do you think? If you disagree I am ok to merge this as-is.

**Comment by non** [comment]
:+1: 

You might want to leave a comment explaining why aliasing `ap` to `apply` is useful. (Namely that the class' method shadows the smart constructor, right?)

**Comment by adelbertc** [comment]
Comment added - I think the comment is enough, naming the smart constructor `ap` is consistent with the operation it's performing.

**Comment by non** [comment]
Great. :+1:

**Comment by non** [comment]
:+1:

**Comment by ceedubs** [comment]
Nice! :+1:

**Pull request merged**